### ISSUE TITLE
Shape Mismatches between predictions and labels

### ISSUE THREAD

**Issue by HariWu1995** [PaddlePaddle/PaddleOCR#7315]
Hi, I am trying to simplify **PGNet** end-to-end by extracting only necessary functions and classes from your repo. But I am having shape mismatches when calculating loss.

After I run dataloader, I have shape as below:
- image
`torch.Size([16, 3, 512, 512])`
- tcl_maps
`torch.Size([16, 1, 1, 1])`
- tcl_label_maps
`torch.Size([16, 1, 25, 25])`
- border_maps
`torch.Size([16, 5, 25, 25])`
- direction_maps
`torch.Size([16, 3, 25, 25])`
- training_masks
`torch.Size([16, 1, 25, 25])`
- label_list
`torch.Size([16, 42, 22, 1])`
- pos_list
`torch.Size([16, 42, 64, 3])`
- pos_mask
`torch.Size([16, 42, 64, 1])`

After feeding these inputs through model, I get outputs as below:
- f_center
`torch.Size([16, 1, 12, 12])`
- f_border
`torch.Size([16, 4, 12, 12])`
- f_direction
`torch.Size([16, 2, 12, 12])`
- f_character
`torch.Size([16, 210, 12, 12])`

I check the loss function and see there are mismatches between predictions and labels:
- f_center vs tcl_maps vs training_masks
- f_border vs border_maps vs tcl_maps vs training_masks
- f_direction vs direction_maps vs tcl_maps vs training_masks
- f_character vs pos_list vs pos_mask vs label_list

Can you share the expected shape of inputs and outputs? Because I cannot find information of shape in the paper.

**Comment by wangjingyeye** [comment]
If you use the original version of pgnet, the shape of these features should be:

1. input：
- image
  torch.Size([16, 3, 512, 512])
- tcl_maps
  torch.Size([16, 1, 128, 128])
- tcl_label_maps
  torch.Size([16, 1, 128, 128])
- border_maps
  torch.Size([16, 5, 128, 128])
- direction_maps
  torch.Size([16, 3, 128, 128])
- training_masks
  torch.Size([16, 1, 128, 128])
- label_list
  torch.Size([16, 30, 50, 1])
- pos_list
  torch.Size([16, 30, 64, 3])
- pos_mask
  torch.Size([16, 30, 64, 1])

2. output
- f_center
- torch.Size([16, 1, 128, 128])
- f_border
- torch.Size([16, 4, 128, 128])
- f_direction
- torch.Size([16, 2, 128, 128])
- f_character
- torch.Size([16, 37, 128, 128])

**Comment by HariWu1995** [comment]
Thanks a lot for your support. I think I miss something when use your PGNet processor. I will debug this and come to you if I need anything else.
